### PR TITLE
Tech 7916 add graphite usage logging

### PR DESCRIPTION
**Pull request by jebentier**

### CODE REVIEW

File: lib/invoca/metrics/client.rb [hunk 1]
```diff
@@ -81,6 +81,7 @@ def server_name # For backwards compatibility
       # This will store the gauge value passed in so that it is reported every GAUGE_REPORT_INTERVAL
       # seconds and post the gauge at the same time to avoid delay in gauges being
       def gauge(name, value)
+        log_usage(name, :guage)
```
Comment (dcaddell) [inline lib/invoca/metrics/client.rb:84]:
```suggestion
        log_usage(name, :gauge)
```

File: lib/invoca/metrics/client.rb [hunk 2]
```diff
@@ -89,6 +90,7 @@ def gauge(name, value)
 
       def count(name, value = 1)
         if (args = normalized_metric_name_and_value(name, value, "counter"))
+          log_usage(args.first, :counter)
```
Comment (dcaddell) [inline lib/invoca/metrics/client.rb:93]: Why is this the only place we use the normalized metric name for logging?

File: lib/invoca/metrics/client.rb [hunk 3]
```diff
@@ -146,6 +150,20 @@ def normalized_metric_name_and_value(name, value, stat_type)
           [extended_name]
         end
       end
+
+      def log_usage(metric_name, metric_type)
+        call_stack = caller_locations
+        unless call_stack.any? { |l| /invoca\/metrics\/prometheus/.match?(l.to_s) }
+          self.class.logger&.info(
+            "Deprecated usage of grapihite metrics",
```
Comment (dcaddell) [inline lib/invoca/metrics/client.rb:158]:
```suggestion
            "Deprecated usage of graphite metrics",
```

File: lib/invoca/metrics.rb [hunk 4]
```diff
@@ -18,7 +18,14 @@ module Metrics
 
     class << self
       attr_accessor(*(CONFIG_FIELDS - [:service_name]), :default_config_key)
-      attr_writer :service_name
+      attr_writer :service_name, :graphite_usage_logging_enabled
```
Comment (dcaddell) [inline lib/invoca/metrics.rb:21]: This doesn't look like it's being used.

**Review by dcaddell — CHANGES_REQUESTED** [review]
This approach lgtm, just a couple small comments. I like that call stack trick that the logger is using.

File: lib/invoca/metrics/client.rb (hunk 2 above)
Comment (jebentier) [reply lib/invoca/metrics/client.rb]: This should be `name` like the others now. Thanks for catching this.

**Review by robertp — APPROVED** [review]
clean!

**Pull request merged**